64_allowlist_signing.md

```diff
@@ -0,0 +1,290 @@
+
+### Goals
+
+<!--
+List the specific goals of the enhancement.  What is it trying to achieve?  How will we
+know that this has succeeded?
+-->
+
+Users should be able to sign allowlists using new, modern key algorithms (such as elliptic curves) in addition to GPG. The user should also have an option to ask the tenant to perform additional verification of allowlist metadata against a transparency log (either online or offline for air gapped machines).
+
+### Non-Goals
+
+<!--
+What is out of scope for this enhancement?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Integrated signing of Keylime allowlists
+- Additional allowlist verification off-of-tenant
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+Currently, the Keylime tenant performs a signature verification step whenever an allowlist is loaded and a signature file and public key are provided as input. This proposal suggests that:
+
+- a new option should be added to this verification step that would also check the provided signature and public key for inclusion in a transparency log
+- more modern key types (such as ECDSA) should be accepted during the signature verification step
+
+
+### User Stories (optional)
+
+<!--
+Detail the things that people will be able to do if this enhancement is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+- A Keylime deployment with access available to a transparency log (can be deployed privately or a public instance is available and run by the Linux Foundation).
+- User provides the tenant with paths to an allowlist, a signature, and a public key using a modern algorithm (e.g. ECDSA)
+- Keylime verifies the inputs are consistent, then checks Rekor for an inclusion proof of those artifacts, before loading them into Keylime
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+enhancement ecosystem.
+
+How will security be reviewed and by whom?
+-->
+
```

Comment:
Since this is a feature for enhanced security, we should add some notes about how the security review will take place.

Reply:
What would that look like? Are there any existing enhancement proposals that have suitable language I could draw on?